### ISSUE TITLE
Interactions mysteriously fail and no events trigger

### ISSUE THREAD

**Issue by Nygosaki** [tandemdude/hikari-lightbulb#412]
### Steps to reproduce

```python
# Import the command handler
import lightbulb

# Instantiate a Bot instance
bot = lightbulb.BotApp(token="<secret>", prefix="!")

@bot.command
@lightbulb.command("ping", "checks the bot is alive")
@lightbulb.implements(lightbulb.SlashCommand)
async def ping(ctx: lightbulb.Context) -> None:
    # Send a message to the channel the command was used in
    await ctx.respond("Pong!")

bot.run()
```

I am running the code on macOS in a venv within visual studio code.

### Expected result

Events, such as `@bot.listen()` as well as slash commands should function.

### Actual result

Bot comes online, updates the list of slash commands, the activity is set if one is specified, but events or slash commands dont work. I have tested a discord.js implementation as well which has worked, meaning that it is an issue with hikari-lightbulb.

### System info

```shell
hikari-lightbulb (2.3.3)
hikari (2.0.0.dev122) [5961d1a3]
located at /Users/nygosaki/Development/Personal/general-discord-app/.venv/lib/python3.11/site-packages/hikari
CPython 3.11.5 Clang 13.0.0 (clang-1300.0.29.30)
Darwin Milos-Macy-Pro-3.local 23.3.0 Darwin Kernel Version 23.3.0: Wed Dec 20 21:31:00 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T6020 arm64 armd
```

### Checklist

- [X] I have made sure to remove ANY sensitive information (bot token, passwords, credentials, personal details, etc.).
- [X] I have searched the issue tracker and have made sure it's not a duplicate. If it is a follow up of another issue, I have specified it.

**Comment by null-domain** [comment]
I ran the code you provided verbatim and the `ping` command worked as expected. Make fully sure your code has been saved in VS Code before running (IIRC it doesn't save automatically by default).

Some events require privileged intents to be both enabled in your [Bot Dashboard](https://discord.com/developers/applications) and passed into the BotApp constructor; if you're using any events that require privileged intents you will need to do this.

Closing as I could not identify an issue with Lightbulb; if you can prove otherwise please feel free to reopen.

**Comment by Nygosaki** [comment]
> I ran the code you provided verbatim and the `ping` command worked as expected. Make fully sure your code has been saved in VS Code before running (IIRC it doesn't save automatically by default).
> 
> Some events require privileged intents to be both enabled in your [Bot Dashboard](https://discord.com/developers/applications) and passed into the BotApp constructor; if you're using any events that require privileged intents you will need to do this.
> 
> Closing as I could not identify an issue with Lightbulb; if you can prove otherwise please feel free to reopen.

My code has been properly saved. In fact, I have tried 3 different Hikari-lightbulb scripts that used to work in the past. It can not be an issue with permissions or discord, as when using discord.js the bot worked.    
I suspect that it may have to do something with how macOS handles things. Did you attempt to replicate the issue on a macOS Sonoma (build 23D60) device?

https://github.com/tandemdude/hikari-lightbulb/assets/63132234/a540b066-78d1-492d-9414-45aa4ab71331

(don't worry, the token is reset now)

**Comment by Nygosaki** [comment]
Also, I am unable to reopen the issue, thus I am resorting to tagging you instead @null-domain

**Comment by tandemdude** [comment]
I am running the code on a Macbook - the code you sent works perfectly. I suspect your discord client has not updated its version of the slash command (hence causing the interaction to fail **NOT** the bot not to respond) or that you are trying to run the guild-specific slash command instead of the global one that the script would create.

I would recommend setting the logging to debug mode to see what is going wrong and if the bot is even receiving the interaction. This is not an issue with lightbulb as the given code works perfectly.

**Comment by Nygosaki** [comment]
> I am running the code on a Macbook - the code you sent works perfectly. I suspect your discord client has not updated its version of the slash command (hence causing the interaction to fail **NOT** the bot not to respond) or that you are trying to run the guild-specific slash command instead of the global one that the script would create.
> 
> I would recommend setting the logging to debug mode to see what is going wrong and if the bot is even receiving the interaction. This is not an issue with lightbulb as the given code works perfectly.

This is weird.   
With the updating of the slash command, my bad. In the test code I used I forgot to set only a specific guild (to speed up the slash command propagation), but the same issue occurs even when the slash command is properly set.

Current code:
```python
# Import the command handler
import lightbulb
import hikari
import os

# Instantiate a Bot instance
bot = lightbulb.BotApp(token=str(os.environ["BOT_TOKEN"]), logs="DEBUG", intents=hikari.Intents.ALL, default_enabled_guilds=[int(os.environ["GUILD_ID"])])

# Register the command to the bot
@bot.command
# Use the command decorator to convert the function into a command
@lightbulb.command("ping", "checks the bot is alive")
# Define the command type(s) that this command implements
@lightbulb.implements(lightbulb.SlashCommand)
# Define the command's callback. The callback should take a single argument which will be
# an instance of a subclass of lightbulb.Context when passed in
async def ping(ctx: lightbulb.Context) -> None:
    # Send a message to the channel the command was used in
    await ctx.respond("Pong!")

@bot.listen()
async def messageLog(event: hikari.GuildMessageCreateEvent) -> None:
    print(event.message)

# Run the bot
# Note that this is blocking meaning no code after this line will run
# until the bot is shut off
bot.run()
```

Output:   
```
Thank you for using lightbulb!
W 2024-04-20 15:45:43,475 hikari.ux: You are running on optimization level 0 (no optimizations), which may slow down your application. For production, consider using at least level 1 optimization by passing `-O` to the python interpreter call
D 2024-04-20 15:45:43,493 hikari.event_manager: subscribing callback 'async def _manage_application_commands(_: 'hikari.StartingEvent') -> 'None'' to event-type hikari.events.lifetime_events.StartedEvent
D 2024-04-20 15:45:43,493 hikari.event_manager: subscribing callback 'async def handle_interaction_create_for_application_commands(event: 'hikari.InteractionCreateEvent') -> 'None'' to event-type hikari.events.interaction_events.InteractionCreateEvent
D 2024-04-20 15:45:43,493 hikari.event_manager: subscribing callback 'async def handle_interaction_create_for_autocomplete(event: 'hikari.InteractionCreateEvent') -> 'None'' to event-type hikari.events.interaction_events.InteractionCreateEvent
D 2024-04-20 15:45:43,493 lightbulb.app: Registering command 'ping'. Requested types are: SlashCommand
D 2024-04-20 15:45:43,493 hikari.event_manager: subscribing callback 'async def messageLog(event: hikari.events.message_events.GuildMessageCreateEvent) -> None' to event-type hikari.events.message_events.GuildMessageCreateEvent
D 2024-04-20 15:45:43,493 asyncio: Using selector: KqueueSelector
D 2024-04-20 15:45:43,494 hikari.ratelimits: GET /gateway/bot is being mapped to new bucket UNKNOWN;<censored>;<censored>
I 2024-04-20 15:45:43,609 hikari.ux: A newer version of hikari is available, consider upgrading to 2.0.0.dev124
D 2024-04-20 15:45:43,844 hikari.ratelimits: remapping GET /gateway/bot with existing bucket UNKNOWN;<censored>;<censored> [reset-after:5.0s, limit:2, remaining:1]
I 2024-04-20 15:45:43,845 hikari.bot: you can start 995 sessions before the next window which starts at 2024-04-21 15:22:02.797088+02:00; planning to start 1 session... 
D 2024-04-20 15:45:44,000 hikari.gateway.0: identifying with new session
D 2024-04-20 15:45:44,009 hikari.gateway.0: starting heartbeat with interval 41.25s
I 2024-04-20 15:45:44,251 hikari.gateway.0: shard is ready: 1 guilds, <censored> (<censored>), session '<censored>' on v10 gateway
D 2024-04-20 15:45:44,251 hikari.cache: setting my user to <censored>
D 2024-04-20 15:45:44,252 hikari.bot: shard 0 started successfully in 407.2ms
D 2024-04-20 15:45:44,253 hikari.ratelimits: GET /oauth2/applications/@me is being mapped to new bucket UNKNOWN;<censored>;<censored>
D 2024-04-20 15:45:44,585 hikari.ratelimits: remapping GET /oauth2/applications/@me with existing bucket UNKNOWN;<censored>;<censored> [reset-after:0.001s, limit:1000, remaining:999]
I 2024-04-20 15:45:44,587 lightbulb.internal: Processing guild application commands
D 2024-04-20 15:45:44,588 hikari.ratelimits: GET /applications/<censored>/guilds/<censored>/commands is being mapped to new bucket UNKNOWN;<censored>;<censored>
D 2024-04-20 15:45:44,812 hikari.ratelimits: remapping GET /applications/<censored>/guilds/<censored>/commands with existing bucket UNKNOWN;<censored>;<censored> [reset-after:0.001s, limit:1000, remaining:999]
I 2024-04-20 15:45:44,814 lightbulb.internal: Processing application commands for guild <censored>
D 2024-04-20 15:45:44,814 lightbulb.internal: <censored> - Created: 0, Deleted: 0, Updated: 1, Unchanged: 0, Skipped: 0
D 2024-04-20 15:45:44,815 hikari.ratelimits: PUT /applications/<censored>/guilds/<censored>/commands is being mapped to new bucket UNKNOWN;<censored>;<censored>
D 2024-04-20 15:45:45,029 hikari.ratelimits: remapping PUT /applications/<censored>/guilds/<censored>/commands with existing bucket UNKNOWN;<censored>;<censored> [reset-after:30.0s, limit:2, remaining:1]
I 2024-04-20 15:45:45,030 lightbulb.internal: Processing global application commands
D 2024-04-20 15:45:45,030 hikari.ratelimits: GET /applications/<censored>/commands is being mapped to new bucket UNKNOWN;<censored>;<censored>
D 2024-04-20 15:45:45,266 hikari.ratelimits: remapping GET /applications/<censored>/commands with existing bucket UNKNOWN;<censored>;<censored> [reset-after:0.001s, limit:1000, remaining:999]
D 2024-04-20 15:45:45,267 lightbulb.internal: Global - Created: 0, Deleted: 0, Updated: 0, Unchanged: 0, Skipped: 0
I 2024-04-20 15:45:45,267 lightbulb.internal: Application command processing completed
I 2024-04-20 15:45:45,267 hikari.bot: started successfully in approx 1.77 seconds
D 2024-04-20 15:46:03,496 hikari.ratelimits: <censored>;<censored>;- rate limiter closed
D 2024-04-20 15:46:03,496 hikari.ratelimits: <censored>;<censored>;- rate limiter closed
D 2024-04-20 15:46:03,496 hikari.ratelimits: <censored>;<censored> rate limiter closed
D 2024-04-20 15:46:03,496 hikari.ratelimits: <censored>;<censored>;- rate limiter closed
D 2024-04-20 15:46:03,496 hikari.ratelimits: purged 4 stale buckets, 0 remain in survival, 1 active
Message(id=<censored>, channel_id=<censored>, guild_id=<censored>, author=UserImpl(id=654445577605611531, discriminator='0', username='nygosaki', global_name='Fishy <3', is_bot=False, is_system=False, flags=<UserFlag.HYPESQUAD_BRAVERY|ACTIVE_DEVELOPER: 4194368>), flags=<MessageFlag.NONE: 0>)
```

Video showing the issue:

https://github.com/tandemdude/hikari-lightbulb/assets/63132234/5bc461a8-35c4-41a5-a743-39dfd154ddd3

Video showing discord.js:

https://github.com/tandemdude/hikari-lightbulb/assets/63132234/f2371af5-7dc0-4b9a-b6f1-070a4a0b798b

(yes, the token is invalid now)    
why does it work on your device? I have no idea.   
I have also tried using a couple separate networks, and restarting my computer
